processor/src/main/java/org/mapstruct/ap/internal/model/AbstractMappingMethodBuilder.java

```diff
@@ -73,7 +73,7 @@ Assignment forgeMapping(SourceRHS sourceRHS, Type sourceType, Type targetType) {
             true
         );
 
-        return createForgedBeanAssignment( sourceRHS, forgedMethod );
+        return createForgedAssignment( sourceRHS, forgedMethod );
```

Comment:
Hey guys.. I was thinking. Should forging not be part of the selector or `MappingResolverImpl`? So, if you cannot select something else.. Just forge a method. Then it would solve it not only for `BeanMapping`, but also `IterableMapping` en `MapMapping`..